### PR TITLE
Refactored daemon-scheduler environment and deployment structure.

### PR DESCRIPTION
**Pull request by kylbarnes**
#### Summary
Refactored daemon-scheduler environment and deployment structure.

#### Implementation details
* Moved environment to daemon-scheduler/pkg/environment
* Moved types.Deployment to daemon-scheduler/pkg/deployment/types
* Renamed deployment.Environment to environment.EnvironmentService
* Renamed deployment.Deployment to deployment.DeploymentService

#### Testing
- [x] cluster-state-service binary built locally and unit-tests pass (`cd cluster-state-service; make; cd ../`)
- [x] cluster-state-service build in Docker succeeds (`cd cluster-state-service; make release; cd ../`)
- [x] daemon-scheduler binary built locally and unit-tests pass (`cd daemon-scheduler; make; cd ../`)
- [x] daemon-scheduler build in Docker succeeds (`cd daemon-scheduler; make release; cd ../`)

New tests cover the changes: no

#### Description for the changelog
Refactored daemon-scheduler environment and deployment structure.

#### Licensing
This contribution is under the terms of the Apache 2.0 License: Yes

#### Before merging
- [x] daemon-scheduler integration tests pass. Required setup details are listed [here](https://github.com/blox/blox/blob/dev/daemon-scheduler/internal/features/README.md).
- [x] daemon-scheduler end-to-end tests pass. Required setup details are listed [here](https://github.com/blox/blox/blob/dev/daemon-scheduler/internal/features/README.md).

### CODE REVIEW

File: daemon-scheduler/pkg/api/v1/api_test.go [hunk 1]
```diff
@@ -282,7 +283,7 @@ func (suite *APITestSuite) TestDeleteEnvironmentMissingEnvironment() {
 	assert.Equal(suite.T(), http.StatusNotFound, responseRecorder.Code)
 }
 
-func (suite *APITestSuite) assertSame(environment *types.Environment, environmentModel *models.Environment) {
+func (suite *APITestSuite) assertSame(environment *environmenttypes.Environment, environmentModel *models.Environment) {
```
Comment (tuedtran) [inline daemon-scheduler/pkg/api/v1/api_test.go]: Shouldn't it be environmentTypes? camelCase?

Comment (kylbarnes) [inline daemon-scheduler/pkg/api/v1/api_test.go]: Yes for variables. No for packages. environmenttypes is a package, not a variable. You'll see we reference package names in all lowercase everywhere.

Comment (tuedtran) [inline daemon-scheduler/pkg/api/v1/api_test.go]: Oh I see

File: daemon-scheduler/pkg/engine/scheduler.go [hunk 2]
```diff
@@ -37,20 +39,20 @@ const (
...
-	deploymentSvc  deployment.Deployment
-	css            facade.ClusterState
-	ecs            facade.ECS
-	events         chan<- Event
-	executionState map[string]environmentExecutionState
-	inProgress     bool
-	inProgressLock sync.RWMutex
+	id                 string
+	ctx                context.Context
+	environmentService environment.EnvironmentService
```
Comment (poojamaiya) [inline daemon-scheduler/pkg/engine/scheduler.go]: Can we make this `environmentSvc` to keep it consistent with `deploymentSvc` declared below? Or vice versa (use `..Service` for both).

Comment (kylbarnes) [inline daemon-scheduler/pkg/engine/scheduler.go]: Yup, I'll rename `deploymentSvc` to `deploymentService` when I do the deployments refactor.

**Pull request merged**